File: python/docs_code_snippets/knowledge-upload.py
import time
from opperai import Opper

opper = Opper()

# Setup
kb = opper.knowledge.create(name=f"docs-snippet-{int(time.time())}")

# --- docs ---
uploaded = opper.knowledge.upload_file(
    kb.id,
    b"test content",
    filename="document.txt",
    metadata={"source": "upload"},
)
print(f"Uploaded: {uploaded.original_filename} -> {uploaded.document_id}")
# --- /docs ---

# Cleanup
opper.knowledge.delete(kb.id)

File: typescript/docs_code_snippets/knowledge-upload.ts
import { Opper } from "opperai";

const opper = new Opper();

// setup
const kb = await opper.knowledge.create({ name: `docs-snippet-${Date.now()}` });
const kbId = kb.id;
const pdfBlob = new Blob(["Hello world PDF content"], { type: "application/pdf" });
// /setup

// --- docs ---
const uploaded = await opper.knowledge.uploadFile(kbId, pdfBlob, {
  filename: "document.pdf",
  metadata: { source: "upload" },
});
console.log(`Uploaded: ${uploaded.original_filename} -> ${uploaded.document_id}`);
// --- /docs ---

// cleanup
await opper.knowledge.delete(kbId);
